### DISCUSSION TITLE
Fail on findings, while continuing and printing output

### DISCUSSION THREAD

**Discussion by katexochen** [sharkdp/fd#1438] [Q&A]
A common situation I find myself in is using fd for writing some check, i.e. in CI. For such jobs, I'd like to exit with an error code if I have a match. However, I also would like to get all matches and the output printed. I couldn't find a nice way of doing this with other commands yet, any ideas?

**Comment by tavianator** [comment]
There is a way to delay an error to the end of execution with `find`: the `-exec command ... {} +` syntax will cause `find` to exit with a non-zero status if any invocation of the `command` fails, but it won't stop execution right away.  So something like this:

```console
$ find -name '*err*' -print -exec false {} +
./src/error.rs
./target/release/build/anyhow-50b078fd38fb554c/stderr
...
$ echo $?
1
```

Actually now that I think of it, you can do the same thing with `fd -X`, it's just a little more effort to have it also print the results:

```console
$ fd err -X printf '%s\n' {} \; -X false {} \;
./src/error.rs
./target/release/build/jemalloc-sys-45a83dd636cff359/stderr
...
$ echo $?
1
```

**Reply by katexochen** [reply]
Thanks, `-X` is a good idea. However, I'm using `--exec` in my commands, which cannot be combined with `--exec-batch` (not sure why).

**Reply by tavianator** [reply]
Ah, then you may be stuck with `find`.  Something like

```console
$ find -name '*foo*' -exec false {} + -exec bar {} \;
```

**Reply by katexochen** [reply]
Might the `+` semantic of find be worth a feature request?

**Reply by tavianator** [reply]
The `{} +` semantic of `find` is just `-X`.  It might be a reasonable feature request to allow combining `-x` with `-X`.  I'm not sure why that's forbidden.
